### PR TITLE
Zgc/diopi ascend add norm op

### PR DESCRIPTION
**Pull request by zhaoguochun1995**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation and Context

1. 添加了基于op-plugin 随机数生成相关算子的基础设施
2. 添加了diopiNormal、diopiNormalInp算子
3. 添加了TensorWrapper类，后期希望劫持at::Tensor的部分成员函数

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/ascend_npu/torch_npu/csrc/framework/DIOPIAdapter.h [hunk 1]
```diff
@@ -16,6 +16,38 @@
...
+
+    // fix: invalid initialization of reference of type 'const at::ascend_npu::TensorWrapper&' from expression of type 'const at::Tensor'
+    explicit TensorWrapper(const at::Tensor& other) : at::Tensor::Tensor(other) {}
+
+    // fix: have different types 'const at::ascend_npu::TensorWrapper' and 'at::Tensor'
+    operator at::Tensor() const { return static_cast<at::Tensor>(*this); }
+
+    // fix: no match for 'operator=' (operand types are 'at::ascend_npu::TensorWrapper' and 'at::Tensor')
+    TensorWrapper operator=(at::Tensor other) {
+        at::Tensor::operator=(other);
```
Comment (yangbofun) [inline impl/ascend_npu/torch_npu/csrc/framework/DIOPIAdapter.h:36]: TensorWrapper& operator=(const at::Tensor& other){
    if (&other == this){
        return *this
    }
......


}

Comment (zhaoguochun1995) [reply impl/ascend_npu/torch_npu/csrc/framework/DIOPIAdapter.h:36]: 这里只是薄薄的一层，直接调用at::Tensor::operator=()就好。

Comment (yangbofun) [reply impl/ascend_npu/torch_npu/csrc/framework/DIOPIAdapter.h:36]: 最好还是遵循operator=的规范写法，返回引用，形参为引用。对于是否是&other==this，充分相信at::Tensor的话，不写也可以，只是少了上层的检查，交给了at::Tensor来做。

Comment (yangbofun) [reply impl/ascend_npu/torch_npu/csrc/framework/DIOPIAdapter.h:36]: > 这里只是薄薄的一层，直接调用at::Tensor::operator=()就好。

最好还是遵循operator=的规范写法，返回引用，形参为引用。对于是否是&other==this，充分相信at::Tensor的话，不写也可以，只是少了上层的检查，交给了at::Tensor来做

**Review by yangbofun — APPROVED** [review]

File: impl/ascend_npu/torch_npu/csrc/framework/DIOPIAdapter.h (hunk 1 above)
Comment (yangbofun) [reply impl/ascend_npu/torch_npu/csrc/framework/DIOPIAdapter.h:36]: 下个pr修改。

File: impl/ascend_npu/diopi_impl/helper.hpp [hunk 2]
```diff
@@ -284,9 +299,9 @@ template <>
 inline std::string dumpArgs(const at::Tensor& t) {
     std::stringstream stream;
     if (t.defined()) {
-        stream << " shape:" << t.sizes() << ", t.strides:" << t.strides() << t.options() << ",ptr:" << t.data_ptr();
+        stream << " shape:" << t.sizes() << ", numel:" << t.numel() << ", t.strides:" << t.strides() << t.options() << ",ptr:" << t.data_ptr();
```
Comment (jingguo-st) [inline impl/ascend_npu/diopi_impl/helper.hpp:302]: t.strides() << t.options()少一个提示字符串，两个值会连在一起

**Pull request merged**